lib/mongodb.js

```diff
@@ -141,7 +141,7 @@ module.exports = class MongoDB {
 			const res = await this.mongo.makeQuery(model, collection => collection
 				.findOneAndUpdate(filters, operationGroupedValues, { upsert: true, returnNewDocument: true }));
 
-			return res && res.value && res.value._id.toString();
+			return res.value ? res.value._id.toString() : res.lastErrorObject.upserted.toString();
```

Comment:
Cuidado que tampoco es 100% seguro que `upserted` exista, asi que recomiendo

```suggestion
			return res.value ? res.value._id.toString() : res.lastErrorObject.upserted?.toString();
```